cpp/src/arrow/engine/substrait/relation_internal.h

```diff
@@ -40,9 +40,46 @@ struct DeclarationInfo {
   int num_columns;
 };
 
+/// \brief A function to extract Acero Declaration from a Substrait Rel object
 ARROW_ENGINE_EXPORT
 Result<DeclarationInfo> FromProto(const substrait::Rel&, const ExtensionSet&,
                                   const ConversionOptions&);
 
+/// \brief Serializes a Declaration, produce a Substrait Rel and update the global
+/// Substrait plan. A Substrait Rel is passed as a the plan and it is updated with
+/// corresponding Declaration passed for serialization.
+///
+/// Note that this is a rather a helper method useful to fuse a partially serialized
+/// plan with another plan. The reason for having a partially serialized plan is to
+/// avoid unnecessary complication and enable partial plan serialization without
+/// affecting a global plan. Since kept as unique_ptr resources are relased efficiently
+/// upon releasing for the global plan.
+ARROW_ENGINE_EXPORT Status SerializeAndCombineRelations(const compute::Declaration&,
+                                                        ExtensionSet*,
+                                                        std::unique_ptr<substrait::Rel>&,
```

Comment:
```suggestion
                                                        std::unique_ptr<substrait::Rel>*,
```
Arrow's style guide forbids the use of mutable references for out-parameters.  You should use a pointer here.

Reply:
Thanks for noting this.